src/context.js

```diff
@@ -1,10 +1,18 @@
 "use strict";
 module.exports = function(Promise) {
+
+var util = require("./util");
 var longStackTraces = false;
 var contextStack = [];
 
-Promise.prototype._promiseCreated = function() {};
-Promise.prototype._pushContext = function() {};
+util.wrapMethodIfExistsDefineIfNot(Promise.prototype, "_promiseCreated",
```

Comment:
what if cancellation is enabled, then monitor is enabled, then cancellation is disabled? Then monitor would not work